crates/ruff_linter/src/settings/types.rs

```diff
@@ -600,57 +600,75 @@ impl Display for RequiredVersion {
 /// pattern matching.
 pub type IdentifierPattern = glob::Pattern;
 
+#[derive(Debug, Clone, CacheKey)]
+pub struct CompiledPerFileIgnore {
+    pub absolute_matcher: GlobMatcher,
+    pub basename_matcher: GlobMatcher,
+    pub negated: bool,
+    pub rules: RuleSet,
+}
+
+impl Display for CompiledPerFileIgnore {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln!(
+            f,
+            "{{ absolute_matcher = {:#?}, \
+                basename_matcher = {:#?}, \
+                negated = {}, \
+                rules = {} }}",
+            self.absolute_matcher, self.basename_matcher, self.negated, self.rules
+        )
+    }
+}
```

Comment:
I think we have a custom macro for displaying settings. 

https://github.com/astral-sh/ruff/blob/317d2e4c757f175e275eeb077f4d2045c8797956/crates/ruff_workspace/src/settings.rs#L57-L76

@snowsignal should we use the `display_settings` macro here?

Reply:
Yep, this would be a good use case for it 😄

Reply:
Paired with Alex to implement this!

Reply:
It did require adding a new format/type whatever of "globmatcher" to the `display_settings!` macro.